src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseTestCaseResource.java

```diff
@@ -86,19 +86,20 @@ public ProgrammingExerciseTestCaseResource(ProgrammingExerciseTestCaseRepository
     }
 
     /**
-     * Update the weights of the provided test case dtos. We don't transfer the whole test case object here, because we need to make sure that only weights can be updated! Will
+     * Update the changable fields of the provided test case dtos. We don't transfer the whole test case object here, because we need to make sure that only weights can be updated! Will
      * only return test case objects in the response that could be updated.
      *
      * @param exerciseId            of exercise the test cases belong to.
-     * @param testCaseWeightUpdates of the test cases to update the weights of.
+     * @param testCaseProgrammingExerciseTestCaseDTOS of the test cases to update the weights and afterDueDate flag of.
      * @return
      */
-    @PatchMapping(value = "programming-exercise/{exerciseId}/update-test-case-weights")
+    @PatchMapping(value = "programming-exercise/{exerciseId}/update-test-cases")
     @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<Set<ProgrammingExerciseTestCase>> updateWeights(@PathVariable Long exerciseId, @RequestBody Set<WeightUpdate> testCaseWeightUpdates) {
-        log.debug("REST request to update the weights {} of the exercise {}", testCaseWeightUpdates, exerciseId);
+    public ResponseEntity<Set<ProgrammingExerciseTestCase>> updateWeights(@PathVariable Long exerciseId,
```

Comment:
I guess this method name should also be updated to reflect that more than the weights can change

Reply:
You are right, I didn't notice. I called it now `updateTestCases`.